### PR TITLE
SLU Multistage - Add Wav2vec2 CTC + Inference

### PR DESCRIPTION
**Pull request by aheba**
Hi, thanks @Moumeneb1  for this PR,

- [ ] add Wav2vec2CTC
- [ ] Load wav2vec2CTC Module from HF and predict text from wav form

### CODE REVIEW

**Comment by aheba** [comment]
This work is related to SLU multi-stage: new perf using Wav2vec2CTC instead of CRDNN. @Moumeneb1 , could you share your numbers

**Comment by TParcollet** [comment]
Hey, can't we decouple their CTC head? I don't like the idea of using their CTC head within SpeechBrain because we already have our.

**Comment by aheba** [comment]
Hey @TParcollet , We use a pre-trained wav2vec2 on Librispeech from HF,  do we have a Librispeech fine-tuned wav2vec within SpeechBrain ?

Or maybe I can convert the output layer ?!

**Comment by TParcollet** [comment]
Let me send you my CV finetuned models (EN).

**Comment by Moumeneb1** [comment]
Hi, 

In the PR we tried to experiment how well can the Wav2vec ASR (trained on CTC) improve the results. 

| Model  | Accuracy|
| ------------- | ------------- |
| CRDNN  LS_LM|  55.55% |
| CRDNN  TAS_LM|  64% |
| Wav2vec as ASR (facebook/wav2vec-base-960h) | 81.02% |
| Wav2vec as ASR (facebook/wav2vec-base-960h) + Bert | 85.53% |

The wav2vec gives better results, it can also be due to the fact that the model is trained with CTC, and doesn't have the looping behaviour as discussed with @lorenlugosch

**Comment by lorenlugosch** [comment]
Cool! If you just use the CRDNN with CTC is it better too? Or is it only if you use wav2vec

**Comment by lorenlugosch** [comment]
I do not. It would probably be easy to swap the original CRDNN encoder into your CTC finetuning script instead of wav2vec, though, if you wanted to try that.

**Comment by TParcollet** [comment]
@aheba https://drive.google.com/file/d/1xOVwifiQNyyUuDT4lHDO9ni2GZSG1ZAz/view?usp=sharing Here is the w2v-en finetuned on commonvoice en (CTC only)

**Comment by mravanelli** [comment]
@aheba, what is the status of this PR?
